servicetalk-http-api/src/main/java/io/servicetalk/http/api/ContentCodingHttpServiceFilter.java

```diff
@@ -126,20 +129,50 @@ public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strat
     }
 
     private static void encodePayloadContentIfAvailable(final HttpHeaders requestHeaders,
+                                                        final HttpRequestMethod method,
+                                                        final HttpProtocolVersion version,
                                                         final List<ContentCodec> supportedEncodings,
                                                         final StreamingHttpResponse response,
                                                         final BufferAllocator allocator) {
         if (supportedEncodings.isEmpty() || hasContentEncoding(response.headers())) {
```

Comment:
nit: add `|| isPassThrough(method, version, response)` to this conditional instead of creating a new `if` block